src/looprestoration.rs

```diff
@@ -1761,3 +1774,212 @@ unsafe fn wiener_filter_neon<BD: BitDepthLooprestorationArm>(
+    const dav1d_sgr_filter_5x5_avx2: looprestorationfilter_fn;
+    const dav1d_sgr_filter_3x3_avx2: looprestorationfilter_fn;
+    const dav1d_sgr_filter_mix_avx2: looprestorationfilter_fn;
+    const dav1d_wiener_filter7_avx512icl: looprestorationfilter_fn;
+    const dav1d_wiener_filter5_avx512icl: looprestorationfilter_fn;
+    const dav1d_sgr_filter_5x5_avx512icl: looprestorationfilter_fn;
+    const dav1d_sgr_filter_3x3_avx512icl: looprestorationfilter_fn;
+    const dav1d_sgr_filter_mix_avx512icl: looprestorationfilter_fn;
+}
+
+#[cfg(all(
+    feature = "bitdepth_8",
+    feature = "asm",
+    any(target_arch = "x86", target_arch = "x86_64"),
+))]
+impl BitDepthLoopRestorationAsm for BitDepth8 {
+    const dav1d_wiener_filter7_sse2: looprestorationfilter_fn = dav1d_wiener_filter7_8bpc_sse2;
+    const dav1d_wiener_filter5_sse2: looprestorationfilter_fn = dav1d_wiener_filter5_8bpc_sse2;
+    const dav1d_wiener_filter7_ssse3: looprestorationfilter_fn = dav1d_wiener_filter7_8bpc_ssse3;
+    const dav1d_wiener_filter5_ssse3: looprestorationfilter_fn = dav1d_wiener_filter5_8bpc_ssse3;
+    const dav1d_sgr_filter_5x5_ssse3: looprestorationfilter_fn = dav1d_sgr_filter_5x5_8bpc_ssse3;
+    const dav1d_sgr_filter_3x3_ssse3: looprestorationfilter_fn = dav1d_sgr_filter_3x3_8bpc_ssse3;
+    const dav1d_sgr_filter_mix_ssse3: looprestorationfilter_fn = dav1d_sgr_filter_mix_8bpc_ssse3;
+    const dav1d_wiener_filter7_avx2: looprestorationfilter_fn = dav1d_wiener_filter7_8bpc_avx2;
+    const dav1d_wiener_filter5_avx2: looprestorationfilter_fn = dav1d_wiener_filter5_8bpc_avx2;
+    const dav1d_sgr_filter_5x5_avx2: looprestorationfilter_fn = dav1d_sgr_filter_5x5_8bpc_avx2;
+    const dav1d_sgr_filter_3x3_avx2: looprestorationfilter_fn = dav1d_sgr_filter_3x3_8bpc_avx2;
+    const dav1d_sgr_filter_mix_avx2: looprestorationfilter_fn = dav1d_sgr_filter_mix_8bpc_avx2;
+    const dav1d_wiener_filter7_avx512icl: looprestorationfilter_fn =
+        dav1d_wiener_filter7_8bpc_avx512icl;
+    // With VNNI we don't need a 5-tap version.
+    const dav1d_wiener_filter5_avx512icl: looprestorationfilter_fn =
+        dav1d_wiener_filter7_8bpc_avx512icl;
+    const dav1d_sgr_filter_5x5_avx512icl: looprestorationfilter_fn =
+        dav1d_sgr_filter_5x5_8bpc_avx512icl;
+    const dav1d_sgr_filter_3x3_avx512icl: looprestorationfilter_fn =
+        dav1d_sgr_filter_3x3_8bpc_avx512icl;
+    const dav1d_sgr_filter_mix_avx512icl: looprestorationfilter_fn =
+        dav1d_sgr_filter_mix_8bpc_avx512icl;
+}
+
+#[cfg(all(
+    feature = "bitdepth_16",
+    feature = "asm",
+    any(target_arch = "x86", target_arch = "x86_64"),
+))]
+impl BitDepthLoopRestorationAsm for BitDepth16 {
+    // NOTE: There are no SSE2 versions of the wiener filter for 16bpc so we fall back
+    // to the C version.
+    const dav1d_wiener_filter7_sse2: looprestorationfilter_fn = wiener_c_erased::<Self>;
+    const dav1d_wiener_filter5_sse2: looprestorationfilter_fn = wiener_c_erased::<Self>;
```

Comment:
I think it might be overly confusing to assign the C versions to the sse2 versions.